pallets/substrate-app/src/lib.rs

```diff
@@ -0,0 +1,687 @@
+    use bridge_types::traits::{
+        BalancePrecisionConverter, BridgeAssetLocker, BridgeAssetRegistry, MessageStatusNotifier,
+        OutboundChannel,
+    };
+    use bridge_types::types::{AssetKind, CallOriginOutput, MessageStatus};
+    use bridge_types::{GenericAccount, GenericNetworkId, SubNetworkId, H256};
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::*;
+    use frame_system::{ensure_root, RawOrigin};
+
+    pub type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+
+    pub type AssetIdOf<T> =
+        <<T as Config>::BridgeAssetLocker as BridgeAssetLocker<AccountIdOf<T>>>::AssetId;
+
+    pub type BalanceOf<T> =
+        <<T as Config>::BridgeAssetLocker as BridgeAssetLocker<AccountIdOf<T>>>::Balance;
+
+    pub type AssetNameOf<T> = <<T as Config>::AssetRegistry as BridgeAssetRegistry<
+        AccountIdOf<T>,
+        AssetIdOf<T>,
+    >>::AssetName;
+
+    pub type AssetSymbolOf<T> = <<T as Config>::AssetRegistry as BridgeAssetRegistry<
+        AccountIdOf<T>,
+        AssetIdOf<T>,
+    >>::AssetSymbol;
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        type OutboundChannel: OutboundChannel<SubNetworkId, Self::AccountId, ()>;
+
+        type CallOrigin: EnsureOrigin<
+            Self::RuntimeOrigin,
+            Success = CallOriginOutput<SubNetworkId, H256, ()>,
+        >;
+
+        type MessageStatusNotifier: MessageStatusNotifier<
+            AssetIdOf<Self>,
+            Self::AccountId,
+            BalanceOf<Self>,
+        >;
+
+        type AssetRegistry: BridgeAssetRegistry<Self::AccountId, AssetIdOf<Self>>;
+
+        type AccountIdConverter: Convert<Self::AccountId, GenericAccount>;
+
+        type AssetIdConverter: Convert<AssetIdOf<Self>, GenericAssetId>;
+
+        type BalancePrecisionConverter: BalancePrecisionConverter<
+            AssetIdOf<Self>,
+            BalanceOf<Self>,
+            GenericBalance,
+        >;
+
+        type BridgeAssetLocker: BridgeAssetLocker<Self::AccountId>;
+
+        type WeightInfo: WeightInfo;
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// [network_id, asset_id, sender, recepient, amount]
+        Burned(
+            SubNetworkId,
+            AssetIdOf<T>,
+            T::AccountId,
+            GenericAccount,
+            BalanceOf<T>,
+        ),
+        /// [network_id, asset_id, sender, recepient, amount]
+        Minted(
+            SubNetworkId,
+            AssetIdOf<T>,
+            GenericAccount,
+            T::AccountId,
+            BalanceOf<T>,
+        ),
```

Comment:
I think name fields are more concise:
```suggestion
        Burned(
            network_id: SubNetworkId,
            asset_id: AssetIdOf<T>,
            sender: T::AccountId,
            recipient: GenericAccount,
            amount: BalanceOf<T>,
        ),
        Minted(
            network_id: SubNetworkId,
            asset_id: AssetIdOf<T>,
            sender: GenericAccount,
            recipient: T::AccountId,
            amount: BalanceOf<T>,
        ),
```

Reply:
Are you sure that that should work?

Reply:
It will work with struct-like variants. For example `Burned { ... }`